pkg/util/crd.go

```diff
@@ -0,0 +1,28 @@
+package util
+
+import (
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// this is a simple implementation for now, may need more sophisticated pluralization logic
```

Comment:
this is fine honestly, although I would add a special case where a "ss" or "us" suffix appends "es", as in "ingress"->"ingresses" or "prometheus"->"prometheuses". that should cover pretty much everything

Reply:
good catch

Reply:
I feel like there's a plurals library floating around somewhere

Reply:
It does use under the hood:
https://github.com/kubernetes/apimachinery/blob/1da46c3f5a5b4a0cc756cb6050df0cf6f06b64c2/pkg/api/meta/restmapper.go#L126-L148
lol

Reply:
guess they had the same idea :smile: